docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,605 @@ under the License.
+#### MapState<UK, UV>
+
+This is a type of [keyed state](#keyed-state) that keeps a list of mappings. You can put key-value 
+pairs into the state and retrieve an Iterable over all currently stored mappings. Mappings are added 
+using put(UK, UV) or putAll(Map<UK, UV>). The value associated with a key can be retrieved using get(UK).
+
+#### Non-keyed State
+
+This type of state is bound to one parallel operator instance and is also called [operator state](#operator-state). 
+
+It is possible to work with [managed state](#managed-state) in non-keyed contexts but it is unusual 
+for user-defined functions to need non-keyed state and the interfaces involved would be different. 
+
+This feature is most often used in the implementation of [sources](#source) and [sinks](#sink).
+
+#### Offset
+
+A number identifying how far you are from the beginning of a certain [DataStream](#datastream). 
 
 #### Operator
 
-Node of a [Logical Graph](#logical-graph). An Operator performs a certain operation, which is
-usually executed by a [Function](#function). Sources and Sinks are special Operators for data
+An operator is a node of a [logical graph](#logical-graph). An operator performs a certain operation, 
+which is usually executed by a [function](#function). Sources and sinks are special operators for data
 ingestion and data egress.
 
 #### Operator Chain
 
-An Operator Chain consists of two or more consecutive [Operators](#operator) without any
-repartitioning in between. Operators within the same Operator Chain forward records to each other
-directly without going through serialization or Flink's network stack.
+An operator chain consists of two or more consecutive [operators](#operator) without any
+repartitioning in between. Operators within the same operator chain forward records to each other
+directly without going through serialization or Flink's network stack. This is a useful optimization
+and increases overall throughput while decreasing latency. The chaining behavior can be configured.
+
+#### Operator State
+
+See [non-keyed state](#non-keyed-state).
+
+#### Parallelism 
+
+This is a technique for making programs run faster by performing several computations simultaneously.
 
 #### Partition
 
-A partition is an independent subset of the overall data stream or data set. A data stream or
-data set is divided into partitions by assigning each [record](#Record) to one or more partitions.
-Partitions of data streams or data sets are consumed by [Tasks](#task) during runtime. A
-transformation which changes the way a data stream or data set is partitioned is often called
-repartitioning.
+A partition is an independent subset of the overall [DataStream](#datastream). A DataStream is divided 
+into partitions by assigning each [record](#record) to one or more partitions via keys. Partitions of 
+DataStreams are consumed by [tasks](#task) during runtime. A transformation that changes the way a 
+DataStream is partitioned is often called repartitioning.
```

Comment:
i would drop the rest after the first sentence here. partition is a broad informatics concept, which has different meanings depending on which angle you see it. The sentences after the first one only creates confusion, and i can't think to a better way to rephrase them.